app/src/main/java/com/google/android/sambadocumentsprovider/mount/MountServerActivity.java

```diff
@@ -158,6 +177,27 @@ public void onClick(View view) {
     mConnectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
 
     restoreSavedInstanceState(savedInstanceState);
+
+    startBrowsing();
+
+    int counter = 0;
```

Comment:
Uh... Is this an answer to an interview question?

There is a better answer to this question, by constructing a recursive formula on number of digits. Define f(n) to be number of numbers which number of digits is less than or equal to n. Obviously f(1) = 1. Then we can have f(n) = (the nth digit is not 9) + (the nth digit is 9) = 9 * f(n-1) + 10^(n-1).

As for generic case, you can carefully count number that exceeds powers of 10 with similar technique.

Reply:
Ooooops... Just wrote it there to check sth really quickly and forgot about it :(